pkg/ingester/tailer.go

```diff
@@ -176,9 +176,11 @@ func (t *tailer) processStream(stream logproto.Stream, lbs labels.Labels) []*log
 
 // Returns true if tailer is interested in the passed labelset
 func (t *tailer) isWatchingLabels(lbs labels.Labels) bool {
-	for _, matcher := range t.matchers {
-		if !matcher.Matches(lbs.Get(matcher.Name)) {
-			return false
+	for _, l := range lbs {
+		for _, matcher := range t.matchers {
+			if l.Name == matcher.Name && !matcher.Matches(l.Value) {
```

Comment:
This would work only if `matchers` has all the label names from `lbs`.
E.g if `lbs` is `foo=bar, env=prod` and `matchers` is `app=bazz`, it would return true because the condition here would be false for all the iterations and we return true at the end.
What do you think?

Reply:
Nice catch.

Reply:
Yep I'll add a test making sure we don't do this one again.